inc/thorvg.h

```diff
@@ -1548,6 +1548,36 @@ class TVG_API Text final : public Paint
      */
     static Result load(const std::string& path) noexcept;
 
+    /**
+     * @brief Loads a scalable font data (ttf) from a memory block of a given size.
+     *
+     * ThorVG efficiently caches the loaded font data using the specified @p name as a key.
+     * This means that loading the same fonts again will not result in duplicate operations.
+     * Instead, ThorVG will reuse the previously loaded font data.
+     * If @p copy is set to @c false, the caching mechanism still works, so it is the user's
+     * responsibility to ensure the @p data's validity for as long as needed.
+     *
+     * @param[in] name The name under which the font will be stored and accessible (e.x. in a @p font() API).
+     * @param[in] data A pointer to a memory location where the content of the ttf data is stored.
+     * @param[in] size The size in bytes of the memory occupied by the @p data.
+     * @param[in] mimeType Mimetype or extension of font data - for now only 'ttf' is supported. In case an empty string is provided the loader will be determined automatically.
+     * @param[in] copy If @c true the data are copied into the engine local buffer (default), otherwise they are not.
+     *
+     * @retval Result::Success When succeed.
+     * @retval Result::InvalidArguments In case no name or no data are provided or the @p size is zero or less (except case when @p data = @c nullptr and @p size = zero are passed to unload the font @p name that was previously loaded).
+     * @retval Result::NonSupport When trying to load a file with an unsupported extension.
+     * @retval Result::Unknown If an error occurs at a later stage.
+     *
+     * @warning: It's the user responsibility to release the @p data memory.
+     *
+     * @note To unload the font data loaded using this API, pass the proper @p name and @c nullptr as @p data and zero as @p size.
+     * @note If you are unsure about the MIME type, you can provide an empty value like @c "", and thorvg will attempt to figure it out.
+     * @note Experimental API
+     *
+     * @see Text::font(const char* name, float size, const char* style)
+     */
+    static Result load(const char* name, const char* data, uint32_t size, const std::string& mimeType = "ttf", bool copy = true) noexcept;
```

Comment:
@hermet please note, that by default copy is true - it's because I need the loader to be cached and want to make it easier for the user - can use the def setting

Reply:
@mgrudzinska Please set the default `false` as other apis do. svg loader could pass `true` there.

Reply:
"If @p copy is set to @c false, the caching mechanism still works, so it is the user's responsibility to ensure the @p data's validity for as long as needed."

this is why I forced true. I'll change to false, but caching still has to be active and it has nothing to do with svg loader. is it ok?

Reply:
Please set false. That is intended. Data ownership will be on user side. Also, it would be better if the svg loader could optimize by avoiding the data copy including the Pictures.

Reply:
@mgrudzinska Please see this design; it's the best scenario we are pursuing.
![image](https://github.com/thorvg/thorvg/assets/3711518/1edfa535-fbd5-4a16-9ebb-1328b60d600f)

Reply:
@hermet, ofc I changed. I'm just trying to highlight, that caching is working no matter what the copy value is (so no matter whether the data is shareable no not), this is opposite to other loading methods

from tvgLoader.cpp:
```
//loads fonts from memory - loader is cached (regardless of copy value) in order to access it while setting font
LoadModule* LoaderMgr::loader(const char* name, const char* data, uint32_t size, const string& mimeType, bool copy)
{
    //TODO: add check for mimetype ?
    if (auto loader = _findFromCache(name)) return loader;
```  

in other places we have:
```
LoadModule* LoaderMgr::loader(const uint32_t *data, uint32_t w, uint32_t h, bool premultiplied, bool copy)
{
    //Note that users could use the same data pointer with the different content.
    //Thus caching is only valid for shareable.
    if (!copy) {
        if (auto loader = _findFromCache((const char*)(data), w * h, "raw")) return loader;
    }
```

Reply:
The loader cache is preferred. It's beneficial if we can support it. let's see